include/envoy/stats/timespan.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <chrono>
+
+#include "envoy/common/time.h"
+#include "envoy/stats/stats.h"
+
+namespace Envoy {
+namespace Stats {
+
+/**
+ * An individual timespan that flushes its measured value to a Duration histogram. The initial time
+ * is captured on construction. A timespan must be completed via complete() for it to be stored. If
+ * the timespan is deleted this will be treated as a cancellation.
+ */
+class Timespan {
+public:
+  Timespan(Histogram& histogram)
+      : histogram_(histogram), start_(std::chrono::steady_clock::now()) {}
+
+  virtual ~Timespan() {}
```

Comment:
nit: not needed

Reply:
Good point.  Removed it and made all the methods non-virtual since nothing's inheriting from it.